src/persist-client/src/internal/metrics.rs

```diff
@@ -769,6 +771,8 @@ pub struct CompactionMetrics {
     pub(crate) parts_prefetched: IntCounter,
     pub(crate) parts_waited: IntCounter,
     pub(crate) fast_path_eligible: IntCounter,
+    // WIP probably should be a shard-level metric instead?
```

Comment:
Reasonable, but happy to defer that until / unless we start using this for a second shard.